### PR TITLE
WIP: (SIMP-9398) pkg:signrpms does not sign RPMs on EL8

### PR DESCRIPTION
**Pull request by lnemsick-simp**
- Fixed bug where pkg:signrpms failed to sign RPMs on EL8.
- Mitigated problem where gpg-agent daemon fails to start because
  its socket path is longer than 108 characters.
  - Changed the default location of the GPG keys directory used in the
    pkg:key_prep and pkg:signrpms Rake tasks to <base_dir>/.dev_gpgkeys.
  - Added a SIMP_PKG_build_keys_dir environment variable that overrides
    the default location of the GPG keys directory used in the
    pkg:key_prep and pkg:signrpms Rake tasks.
- Improved pkg:signrpms error handling and reporting.
- Added SIMP_PKG_rpmsign_timeout environment variable that overrides
  default timeout in seconds to wait for an individual RPM signing
  operation to complete.
  - Default timeout is 30 seconds.
  - Most relevant when signing on RPMs on EL8 and the gpg-agent
    started by rpmsign fails to start, but rpmsign does not detect
    the failure and hangs.

SIMP-9398 #close
SIMP-3864 #close

### CODE REVIEW

File: .github/workflows/pr_ghaction.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Run all tests as GitHub Actions
+name: Unit Tests
```
Comment (op-ct) [inline .github/workflows/pr_ghaction.yml:2]:
```suggestion
name: PR Tests
```

This workflow includes spec + 💪acceptance💪 tests.

**Comment by op-ct** [comment]
The workflow file should be named something less generic than `pr_ghaction.yml`

* `pr_tests.yml` ?
* `pr_tests_spec_and_acc.yml` ?
* `pr_tests__rake_helpers.yml` ?

**Comment by lnemsick-simp** [comment]
Tests are failing in GitHub and I cannot recreate the problems.

**Comment by trevor-vaughan** [comment]
@lnemsick-simp So, I did the following:

1. Tried setting the container to `privileged` mode => no difference
2. Added rngd to the underlying VM => no more hangs
3. Hopped into the container and tried running the following command by hand:

```sh
 rpm --define '%_signature gpg' --define '%__gpg %{_bindir}/gpg' --define '%_gpg_name gatekeeper@simp.development.key' --define '%_gpg_path /home/build_user/test-signrpms/.dev_gpgkeys/dev' --define '%_gpg_digest_algo sha256' --define '%_gpg_sign_cmd_extra_args --pinentry-mode loopback --verbose' --resign /home/build_user/test-signrpms/test.rpms/testpackage-1-0.noarch.rpm
```

The result is that GPG is prompting me for a password.

**Comment by lnemsick-simp** [comment]
It should have prompted you for the password for the GPG key, so that is normal.

**Comment by trevor-vaughan** [comment]
New updates work [correctly in GitHub Actions](https://github.com/trevor-vaughan/rubygem-simp-rake-helpers/runs/2022758395?check_suite_focus=true).

**Pull request merged**